### ISSUE TITLE
A new pipeline for inpainting(object removal) is needed!!

### ISSUE THREAD

**Issue by lawsonxwl** [huggingface/diffusers#4836]
**Is your feature request related to a problem? Please describe.**
Thanks for your great work!!
Right now I'm using StableDiffusionInpaintPipeline with fixed prompt(such as "the empty background") to do object removal task, but one thing i've found is that **when the inpaint mask doesn't cover the shadow or hairline, the model will randomly generate objects else in the mask rather than generating pure background,** which is frustrating. I wonder if the inpaint pipeline is trained for removing object ? If not, when will the inpaint model trained for object removal be released(works like LDM inpainting or LaMa)? As this feature can be really useful in practice!

**Describe the solution you'd like**
Plz release the inpaint pipeline specialized for object removal!

**Describe alternatives you've considered**
Give some advice about getting rid of unwanted object generation

unwanted object generation:
![A57CD3D6-DA0C-4FA3-9A88-373C7E21AC9A](https://github.com/huggingface/diffusers/assets/52605055/7e5a4537-cf4a-4dbd-b4b6-ae9995ff64b7)
![F84FF064-F0D9-43DA-83A4-BF4FACB5E910](https://github.com/huggingface/diffusers/assets/52605055/66d54ad9-77a1-43ec-8c61-db7915292198)
![7CB3DC41-76FF-40CF-B32A-1D3D4A309D50](https://github.com/huggingface/diffusers/assets/52605055/669ac226-78bd-4c61-bfcc-b8271df64d53)

**Comment by GitHub1712** [comment]
Would be cool if we could train ourself. I am missing inpaint training code example still very much.

**Comment by lawsonxwl** [comment]
> Would be cool if we could train tasks like that ourself. I am missing inpaint training code example still very much.

By modifying txt2img training code, inpainting training code can be easy to get.. But controlling the object generated by the model can be really hard!

**Comment by GitHub1712** [comment]
How could we do that? How did the inpaint models learned their abilty to continue structures against txt2img models?

**Comment by lawsonxwl** [comment]
> How could we do that? How did the inpaint models learned their abilty to continue structures against txt2img models?

By finetuning on txt2img models and using random masking strategy, Stability AI trained and released the stable-diffusion inpainting model.. You can check it out on diffusers

**Comment by GitHub1712** [comment]
Thank you, I want to train own if possible based on inpainting sdxl, is there any documentation how this was exactly trained and created, training example code?
I am missing that for all inpaint models...

**Comment by donggoing** [comment]
longing for the example code for finetuning, too.

**Comment by sayakpaul** [comment]
@asomoza do we have any examples/resources for this now?

**Comment by asomoza** [comment]
we have now object removal training-free [guides](https://huggingface.co/blog/OzzyGT/diffusers-image-fill) and [models](https://github.com/huggingface/diffusers/pull/9985) but the discussion was also about training code examples for inpainting which we don't have yet.
